### PR TITLE
Adding option to disable/enable program metrics from runtime args or …

### PR DESCRIPTION
**Pull request by CuriousVini**
…preferences

Issue: https://issues.cask.co/browse/CDAP-12070
Build: https://builds.cask.co/browse/CDAP-RUT1264-4
Tested on cluster

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
Can you add a unit-test?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java [hunk 1]
```diff
@@ -212,6 +210,25 @@ protected AbstractContext(Program program, ProgramOptions programOptions, CConfi
     }
   }
 
+  private MetricsCollectionService getMetricsService(CConfiguration cConf, MetricsCollectionService metricsService,
+                                                     Map<String, String> runtimeArgs) {
+    String overrideMetricsService = runtimeArgs.get(Constants.Metrics.EMIT_PROGRAM_CONTAINER_METRICS);
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java]: How about we simplify the logic as

```
    String emitMetricsPreference = runtimeArgs.get(Constants.Metrics.EMIT_PROGRAM_CONTAINER_METRICS);
     // Get the option for emitting metrics from preferences or as a fall back from cconf
     boolean emitMetrics = emitMetricsPreference != null ? Boolean.getBoolean(emitMetricsPreference) : cConf.getBoolean(Constants.Metrics.EMIT_PROGRAM_CONTAINER_METRICS);
     emitMetricsPreference == true ? return metricsService : return new NoOpMetricsCollectionService();
```

**Comment by sreevatsanraman** [comment]
LGTM

**Comment by CuriousVini** [comment]
Build is green.

**Comment by CuriousVini** [comment]
squashing and merging

**Pull request merged**